### PR TITLE
Advanced search provenance with graphql

### PR DESCRIPTION
**Pull request by marshHawk4**

### CODE REVIEW

File: askem.dbml [hunk 1]
```diff
@@ -128,7 +128,7 @@ Table dataset {
   id int [pk]
   name varchar [not null]
   url varchar [not null]
-  description text [not null]
+  description text
```
Comment (fivegrant) [inline askem.dbml:131]: We probably want to hold off on changes to `askem.dbml` until we have some kind of migration flow set up. Would it be possible to revert this old dbml + autogen files? Is this change necessary to make GraphQL Provenance work?

Comment (fivegrant) [reply askem.dbml:131]: cc @mattprintz

**Review by fivegrant — CHANGES_REQUESTED** [review]
I understand we want to request the nodes and not the edges but I think I'm still a bit confused by what's being returned by the GraphQL. Is it all the related nodes or ones that are a single relationship away? Could we fit some of these provenance queries into the objects themselves (e.g. a `related` field on `Model`)?

**Comment by marshHawk4** [comment]
> 

*for what is being returned. So we are returning all connected nodes n(hops) relationships away or until n (limit) number of nodes are found. Basically it will return all nodes of the subgraph. We allow them to filter the search to only return nodes of interest. They can return only one relationship away if they set hops to 1. I am not exposing the ability to return model revision here.

**Review by fivegrant — CHANGES_REQUESTED** [review]
Ran into an issue with model and publication provenance. Issue has been attached
[test.txt](https://github.com/DARPA-ASKEM/data-service/files/10688950/test.txt)

**Review by fivegrant — APPROVED** [review]

**Review by mattprintz — APPROVED** [review]
Looks good.

**Pull request merged**